tests/integration/golang/mlflow/experiment/set_experiment_tag_test.go

```diff
@@ -1,3 +1,208 @@
+			Valid: true,
+		},
+		LastUpdateTime: sql.NullInt64{
+			Int64: time.Now().UTC().UnixMilli(),
+			Valid: true,
+		},
+		LifecycleStage:   models.LifecycleStageActive,
+		ArtifactLocation: "/artifact/location",
+	})
+	assert.Nil(s.T(), err)
+	defer func() {
+		assert.Nil(s.T(), s.fixtures.UnloadFixtures())
+	}()
+
+	// Set tag on experiment
+	SetExperimentTag(s, experiment, "KeyTag1", "ValueTag1")
+	exp, err := s.fixtures.GetExperimentByID(context.Background(), *experiment.ID)
+	assert.Nil(s.T(), err)
+
+	found := false
+	for _, tag := range exp.Tags {
+		if tag.Key == "KeyTag1" && tag.Value == "ValueTag1" {
+			found = true
+			break
+		}
+	}
+	assert.True(s.T(), found, "Expected 'experiment.tags' to contain 'KeyTag1' with value 'ValueTag1'")
+
+	// Update tag on experiment
+	SetExperimentTag(s, experiment, "KeyTag1", "ValueTag2")
+	exp, err = s.fixtures.GetExperimentByID(context.Background(), *experiment.ID)
+	assert.Nil(s.T(), err)
+
+	found = false
+	for _, tag := range exp.Tags {
+		if tag.Key == "KeyTag1" && tag.Value == "ValueTag2" {
+			found = true
+			break
+		}
+	}
+	assert.True(s.T(), found, "Expected 'experiment.tags' to contain 'KeyTag1' with value 'ValueTag1'")
+
+	//test that setting a tag on 1 experiment does not impact another experiment.
+	exp, err = s.fixtures.GetExperimentByID(context.Background(), *experiment1.ID)
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), len(exp.Tags), 0)
+
+	//test that setting a tag on different experiments maintain different values across experiments
+	SetExperimentTag(s, experiment1, "KeyTag1", "ValueTag3")
+	exp, err = s.fixtures.GetExperimentByID(context.Background(), *experiment.ID)
+	assert.Nil(s.T(), err)
+	exp1, err := s.fixtures.GetExperimentByID(context.Background(), *experiment1.ID)
+	assert.Nil(s.T(), err)
+	found = false
+	for _, tag := range exp.Tags {
+		if tag.Key == "KeyTag1" && tag.Value == "ValueTag2" {
+			found = true
+			break
+		}
+	}
+	assert.True(s.T(), found, "Expected 'experiment.tags' to contain 'KeyTag1' with value 'ValueTag2'")
+
+	found = false
+	for _, tag := range exp1.Tags {
+		if tag.Key == "KeyTag1" && tag.Value == "ValueTag3" {
+			found = true
+			break
+		}
+	}
+	assert.True(s.T(), found, "Expected 'experiment.tags' to contain 'KeyTag1' with value 'ValueTag3'")
+
+}
+func (s *SetExperimentTagTestSuite) Test_Error() {
+	var testData = []struct {
+		name    string
+		error   *api.ErrorResponse
+		request *request.SetExperimentTagRequest
+	}{
+		{
+			name:  "EmptyIDProperty",
+			error: api.NewInvalidParameterValueError("Missing value for required parameter 'experiment_id'"),
+			request: &request.SetExperimentTagRequest{
+				ID:    "",
+				Key:   "test_key",
+				Value: "test_value",
+			},
+		},
+		{
+			name:  "EmptyKeyProperty",
+			error: api.NewInvalidParameterValueError("Missing value for required parameter 'key'"),
+			request: &request.SetExperimentTagRequest{
+				ID:    "1",
+				Key:   "",
+				Value: "test_value",
```

Comment:
here you don't need to send `value`, because test will fail on `key`. Just send as minimum as possible in that case it is understand what is going on.